test/bin/scenario.sh

```diff
@@ -58,16 +80,31 @@ sos_report() {
             # skip log files, etc.
             continue
         fi
-        ip=$(cat "${vmdir}/ip")
-        if [ -z "${ip}" ] ; then
+        local -r vmname=$(basename "${vmdir}")
+        local -r ip=$(get_vm_property "${vmname}" ip)
+        if [ -z "${ip}" ]; then
             # skip hosts without NICs
             # FIXME: use virsh to copy sos report files
             continue
         fi
-        # Copy the sos helper for compatibility, it is only available in 4.14 RPMs
-        scp "${ROOTDIR}/scripts/microshift-sos-report.sh" "redhat@${ip}":/tmp
-        ssh "redhat@${ip}" \
-            "[ -f /usr/bin/microshift-sos-report ] && sudo /usr/bin/microshift-sos-report || sudo /tmp/microshift-sos-report.sh ; sudo chmod +r /tmp/sosreport*"
+        # Some scenarios do not start with MicroShift installed, so we
+        # can't rely on the wrapper being there or working if it
+        # is. Copy the script to the host, just in case, along with a
+        # wrapper that knows how to execute it or the installed
+        # version.
+        cat - >/tmp/sos-wrapper.sh <<EOF
+#!/usr/bin/env bash
+if [ -f /usr/bin/microshift-sos-report ]; then
+    sudo /usr/bin/microshift-sos-report
+else
+    sudo chmod +x /tmp/microshift-sos-report.sh
+    sudo PROFILES=network,security /tmp/microshift-sos-report.sh
+fi
+sudo chmod +r /tmp/sosreport*
+EOF
```

Comment:
The /tmp/sos-wrapper.sh should be executable, so chmod +x seems to be missing either on the source or dest host.